### PR TITLE
MULE-19590: Fix Runtime Extension Models to use composable nested elements

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

**Comment by marianogonzalez** [comment]
--validate

**Comment by marianogonzalez** [comment]
--validate

File: modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/api/extension/XmlSdkTypesValueProvider.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.mule.runtime.extension.api.values.ValueBuilder;
+import org.mule.runtime.extension.api.values.ValueProvider;
+import org.mule.runtime.extension.api.values.ValueResolvingException;
+
+import java.util.Set;
+
+/**
+ * Open {@link ValueProvider} which returns the keys for the primitive types.
+ * <p>
+ * This is needed because XML SDKcan also point to types defined in the application's catalog.
```
Comment (elrodro83) [inline modules/extensions-xml-support/src/main/java/org/mule/runtime/extension/api/extension/XmlSdkTypesValueProvider.java:21]: SDKcan -> SDK can

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java [hunk 2]
```diff
@@ -689,7 +684,7 @@ private void declareRoundRobin(ExtensionDeclarer extensionDeclarer) {
     ConstructDeclarer roundRobin = extensionDeclarer.withConstruct("roundRobin")
         .describedAs("Send each message received to the next message processor in a circular list of targets.");
 
-    roundRobin.withRoute("route").withChain();
+    roundRobin.withRoute("route").withMinOccurs(1).withChain();
```
Comment (elrodro83) [inline runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java:687]: isn't this 2? what is the purpose of a round robin with just one route?

File: runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java [hunk 3]
```diff
@@ -442,7 +449,7 @@ public void choice() {
   public void scatterGather() {
     final ConstructModel scatterGatherModel = coreExtensionModel.getConstructModel("scatterGather").get();
 
-
+    // assertOutputsListOfMessages(scatterGatherModel);
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java:452]: remove? add a TODO?

File: runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java [hunk 4]
```diff
@@ -521,10 +530,18 @@ public void parallelForeach() {
     assertThat(targetValue.isRequired(), is(false));
   }
 
+  private void assertOutputsListOfMessages(HasOutputModel model) {
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java:533]: the call to this method is commented

File: runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java [hunk 5]
```diff
@@ -545,6 +562,7 @@ public void async() {
   public void tryScope() {
     final ConstructModel tryModel = coreExtensionModel.getConstructModel("try").get();
 
+    // assertOutputTypes(tryModel, VOID_TYPE, VOID_TYPE);
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java:565]: remove? TODO>

File: runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java [hunk 6]
```diff
@@ -563,9 +581,10 @@ public void tryScope() {
 
   @Test
   public void untilSuccessful() {
-    final ConstructModel tryModel = coreExtensionModel.getConstructModel("untilSuccessful").get();
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java:566]: OMG 🤦🏼

File: runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java [hunk 7]
```diff
@@ -585,9 +604,10 @@ public void untilSuccessful() {
 
   @Test
   public void firstSuccessful() {
-    final ConstructModel tryModel = coreExtensionModel.getConstructModel("firstSuccessful").get();
```
Comment (elrodro83) [inline runtime-extension-model/src/test/java/org/mule/runtime/core/extension/CoreExtensionModelTestCase.java:588]: copy-paste hell :(

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java (hunk 2 above)
Comment (marianogonzalez) [reply runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java:687]: agree with you, but the component allows it in the XSD.

Comment (elrodro83) [reply runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java:687]: add a comment?

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**